numpy/_core/src/umath/string_buffer.h

```diff
@@ -290,21 +290,15 @@ struct Buffer {
             tmp2++;
         }
         while (tmp1.buf < tmp1.after) {
-            if (*tmp1 < 0) {
-                return -1;
-            }
-            if (*tmp1 > 0) {
+            if (*tmp1) {
```

Comment:
I am confused, wouldn't this make:
```
In [1]: "a" == "a\0"
Out[1]: False
``` 
Return `True`?  For NumPy strings, this should be handled already earlier by the length, and for the new strings embedded 0 characters should be supported.

Reply:
I'm learning new things about numpy strings every day. For some reason I thought that this was `True` until now, but I'm clearly wrong:

```
In [1]: import numpy as np

In [2]: np.__version__
Out[2]: '1.26.3'

In [3]: np.str_("a") == np.str_("a\0")
Out[3]: False
```

The scalar repr doesn't include the trailing nulls, which I guess makes sense but definitely added to my confusion here:

```
In [4]: np.str_("a\0")
Out[4]: 'a'

In [5]: repr(np.str_("a\0")) == repr(np.str_("a"))
Out[5]: True
```

Reply:
Although I guess the above is only true for scalars, for array elements trailing nulls are ignored in Numpy 1.26.3:

```
In [12]: np.array([np.str_("a\0")]) == np.array([np.str_("a")])
Out[12]: array([ True])
```